src/patch-events.js

```diff
@@ -371,9 +379,9 @@ export function addEventListener(type, fnOrObj, optionsOrCapture) {
       if (e.eventPhase !== Event.CAPTURING_PHASE && !e.bubbles && e.target !== target && !(target instanceof Window)) {
         return;
       }
-      let ret = (typeof fnOrObj === 'object' && fnOrObj.handleEvent) ?
-        fnOrObj.handleEvent(e) :
-        fnOrObj.call(target, e);
+      let ret = handlerType === 'function' ?
+        fnOrObj.call(target, e) :
+        (fnOrObj.handleEvent && fnOrObj.handleEvent(e));
```

Comment:
You don't need to check for `handleEvent` since you've already excluded objects without this, yeah?

Reply:
It could have been removed that from the object between adding the event listener and the event getting called. WDYT?

Reply:
ok, sure.